.github/actions/setup-go/action.yaml

```diff
@@ -0,0 +1,28 @@
+name: 'Setup golang with master only caching'
+description: 'A composite action that installs golang, but with a caching strategy that only updates the cache on master branch.'
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/setup-go@v5
+      with:
+        go-version-file: 'go.mod'  # Just use whatever version is in the go.mod file
+        cache: ${{ github.ref == 'refs/heads/master' }}
```

Comment:
Note: This is intentional, for now we are effectively disabling caching on any non-master branch to see where we land in terms of caching usage (we only have 10GB limit for all caching).